extensions/machine-learning-services/src/modelManagement/modelPythonClient.ts

```diff
@@ -13,33 +13,89 @@ import * as utils from '../common/utils';
 import { PackageManager } from '../packageManagement/packageManager';
 import * as constants from '../common/constants';
 import * as os from 'os';
+import { ModelParameters } from './interfaces';
 
 /**
- * Service to import model to database
+ * Python client for ONNX models
  */
-export class ModelImporter {
+export class ModelPythonClient {
 
 	/**
-	 *
+	 * Creates new instance
 	 */
 	constructor(private _outputChannel: vscode.OutputChannel, private _apiWrapper: ApiWrapper, private _processService: ProcessService, private _config: Config, private _packageManager: PackageManager) {
 	}
 
-	public async registerModel(connection: azdata.connection.ConnectionProfile, modelFolderPath: string): Promise<void> {
+	/**
+	 * Deploys models in the SQL database using mlflow
+	 * @param connection
+	 * @param modelPath
+	 */
+	public async deployModel(connection: azdata.connection.ConnectionProfile, modelPath: string): Promise<void> {
 		await this.installDependencies();
-		await this.executeScripts(connection, modelFolderPath);
+		await this.executeDeployScripts(connection, modelPath);
 	}
 
 	/**
-	 * Installs dependencies for model importer
+	 * Installs dependencies for python client
 	 */
-	public async installDependencies(): Promise<void> {
+	private async installDependencies(): Promise<void> {
 		await utils.executeTasks(this._apiWrapper, constants.installDependenciesMsgTaskName, [
 			this._packageManager.installRequiredPythonPackages(this._config.modelsRequiredPythonPackages)], true);
 	}
 
-	protected async executeScripts(connection: azdata.connection.ConnectionProfile, modelFolderPath: string): Promise<void> {
+	/**
+	 *
+	 * @param modelPath Loads model parameters
+	 */
+	public async loadModelParameters(modelPath: string): Promise<ModelParameters> {
+		await this.installDependencies();
+		return await this.executeModelParametersScripts(modelPath);
+	}
+
+	private async executeModelParametersScripts(modelFolderPath: string): Promise<ModelParameters> {
+		modelFolderPath = utils.makeLinuxPath(modelFolderPath);
+
+		let scripts: string[] = [
+			'import onnx',
+			'import json',
+			`onnx_model_path = '${modelFolderPath}'`,
+			`onnx_model = onnx.load_model(onnx_model_path)`,
+			`type_map = {
+				onnx.TensorProto.DataType.FLOAT: 'real',
+				onnx.TensorProto.DataType.UINT8: 'tinyint',
+				onnx.TensorProto.DataType.INT16: 'smallint',
+				onnx.TensorProto.DataType.INT32: 'int',
+				onnx.TensorProto.DataType.INT64: 'bigint',
+				onnx.TensorProto.DataType.STRING: 'varchar(MAX)',
```

Comment:
I haven't a clue about onnx, but figured I'd just check whether this ought to be nvarchar (does onnx support unicode/dualwidth character encoding?)

Reply:
hmm not sure, got the mapping from the service team. will  double check 

---
In reply to: [397444805](https://github.com/microsoft/azuredatastudio/pull/9609#discussion_r397444805) [](ancestors = 397444805)